From: Stanislaw Kardach <stanislaw.kardach@cavium.com>
Date: Sat, 14 May 2016 17:24:48 -0700
Subject: [PATCH] Increase open file limit for OpenDaylight

Upstart scripts by default do not run scripts in a shell which results
in limits from `/etc/security/limits.conf` not be taken into account
and results in the default value (4096) to be taken. This may prove
problematic for OpenDaylight during initialization. On Cavium ThunderX
we have noticed OpenFlow controller initialization failures caused by
hitting this limit.
OPNFV is increasing the `nofile` limit to 112640 for OpenStack services
which is why I'm using this value here.

Signed-off-by: Stanislaw Kardach <stanislaw.kardach@cavium.com>
---
 odl_package/ubuntu/opendaylight | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/odl_package/ubuntu/opendaylight b/odl_package/ubuntu/opendaylight
index 6e5ff7b..52a3159 100644
--- a/odl_package/ubuntu/opendaylight
+++ b/odl_package/ubuntu/opendaylight
@@ -14,7 +14,7 @@ env JAVA_OPTS="-server -Xms256M -Xmx2048M -XX:+UnlockDiagnosticVMOptions -XX:+Un
 env OPTS="-Dkaraf.startLocalConsole=false -Dkaraf.startRemoteShell=true"
 env MAIN="org.apache.karaf.main.Main"
 
-
+limit nofile 102400 112640
 chdir /opt/opendaylight
 
 script
